### PR TITLE
Increase coverage

### LINKED ISSUE #303: Increase test coverage of crystal.py, importer.py, importer_crystal.py, molecule.py, structure.py
I am on it (see PR #305).

### PR DESCRIPTION
**Pull request by adrianusler**
Increase test coverage of crystal.py, importer.py, importer_crystal.py, molecule.py, structure.py

I need the [WIP] pull request to make sure that the tests are passing. My local pytest doesn't work at the moment.

Once finished, this closes #303.

## Summary by CodeRabbit

- **New Features**
	- Added the ability to choose the parser library when importing crystal structures.
	- Introduced a new method for parsing POSCAR files.
	- Added a `center_of_mass` property for structures.
- **Enhancements**
	- Improved the structure object's functionalities, including atom addition and removal, and coordinate centering.
- **Bug Fixes**
	- Fixed issues in crystal and molecule structure tests to accommodate new functionalities and assertions.
- **Refactor**
	- Removed unused imports and methods related to POSCAR file creation.
	- Updated existing methods to enhance functionality and performance.

### CODE REVIEW

**Comment by adrianusler** [comment]
Codecov update on current improvements:
- crystal.py: 71.43% to 91.45%
- importer_crystal.py: 62% to 63.11%
- structure.py: 54.55% to 73.86%

File: src/molara/Structure/io/importer_crystal.py [hunk 1]
```diff
@@ -95,73 +95,81 @@ def __init__(
...
+        scale_, latvecs_ = lines[1], lines[2:5]
+        species_, numbers_, mode = lines[5:8]
+        positions_ = lines[8:]
+        try:
+            scale = float(scale_)
+            basis_vectors = [[float(component) for component in robust_split(latvec_)[:3]] for latvec_ in latvecs_]
+            species = robust_split(species_)
+            numbers = [int(num) for num in robust_split(numbers_)]
+            if len(positions_) == sum(numbers) * 2 + 1:
+                positions_ = positions_[0 : sum(numbers)]
```
Comment (ab5424) [inline src/molara/Structure/io/importer_crystal.py:139]: @coderabbitai generate unit tests for this file.

**Comment by Michel-Heinz** [comment]
Is this ready?

**Comment by adrianusler** [comment]
> Is this ready?

I would say yes. I guess I won't continue working on this any time soon, so we might just as well merge it now.

**Review by ab5424 — APPROVED** [review]

**Pull request merged**